benchmark/utilities/openmm_utilities.py

```diff
@@ -86,43 +114,50 @@ def get_atoms_bonded_to_hydrogen(topology):
                 atom_indices.append(b.index)
             else:
                 atom_indices.append(a.index)
-    return atom_indices
+    return list(set(atom_indices))
 
 
 def repartition_hydrogen_mass_connected(topology, system, h_mass=4.0,
                                         mode="scale"  # or "decrement"
                                         ):
     """Set the mass of all hydrogens to h_mass. Reduce the mass of
     all atoms bonded to hydrogens, so that the total mass remains constant.
-
-    Question: how should we do this, exactly?
-    * Should we subtract the same mass from each bonded atom?
-    * Should we proportionally reduce the mass of each bonded atom?
     """
 
-    raise (NotImplementedError())
+    others = get_atoms_bonded_to_hydrogen(topology)
+    hydrogens = get_hydrogens(topology)
+    initial_h_mass = get_sum_of_masses(system, hydrogens) / len(hydrogens)
+    initial_mass_of_others = get_sum_of_masses(system, others)
+    mass_to_remove_from_others = (h_mass - initial_h_mass) * len(hydrogens)
```

Comment:
Do we really want to operate globally on the sum of all masses like this? That lumps `-CH3`, `-CH2-`, and `-CH-`, along with any `-NH-` or `-OH` cases. Is this what the algorithms really do?

Let's also cite the papers for the algorithms we implement here.